src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dwpuploadresponse/DwpUploadResponseMidEventHandlerTest.java

```diff
@@ -0,0 +1,113 @@
+public class DwpUploadResponseMidEventHandlerTest {
+
+    private static final String USER_AUTHORISATION = "Bearer token";
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+    private DwpUploadResponseMidEventHandler handler;
+    @Mock
+    private Callback<SscsCaseData> callback;
+
+
+    @Mock
+    private CaseDetails<SscsCaseData> caseDetails;
+
+    @Mock
+    private UserDetailsService userDetailsService;
+
+    @Mock
+    private CaseDetails<SscsCaseData> caseDetailsBefore;
+
+    private SscsCaseData sscsCaseData;
+
+    private SscsCaseData sscsCaseDataBefore;
+
+    @Before
+    public void setUp() {
+        handler = new DwpUploadResponseMidEventHandler();
+
+        openMocks(this);
+
+        AddNoteService addNoteService = new AddNoteService(userDetailsService);
+
+
+        when(userDetailsService.buildLoggedInUserName(USER_AUTHORISATION)).thenReturn(UserDetails.builder()
+                .forename("Chris").surname("Davis").build().getFullName());
+
+        when(callback.getEvent()).thenReturn(EventType.DWP_UPLOAD_RESPONSE);
+
+        sscsCaseData = SscsCaseData.builder()
+                .ccdCaseId("1234")
+                .benefitCode("002")
+                .issueCode("CC")
+                .dwpFurtherInfo("Yes")
+                .dwpResponseDocument(DwpResponseDocument.builder().documentLink(DocumentLink.builder().documentUrl("a.pdf").documentFilename("a.pdf").build()).build())
+                .dwpEvidenceBundleDocument(DwpResponseDocument.builder().documentLink(DocumentLink.builder().documentUrl("b.pdf").documentFilename("b.pdf").build()).build())
+                .appeal(Appeal.builder().benefitType(BenefitType.builder().code("taxCredit").build()).build())
+                .build();
+
+        sscsCaseDataBefore = SscsCaseData.builder().build();
+
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(callback.getCaseDetailsBefore()).thenReturn(Optional.of(caseDetailsBefore));
+        when(caseDetails.getId()).thenReturn(Long.valueOf(sscsCaseData.getCcdCaseId()));
+        when(caseDetails.getCaseData()).thenReturn(sscsCaseData);
+        when(caseDetailsBefore.getCaseData()).thenReturn(sscsCaseDataBefore);
+
+    }
+
+
+    @Test
+    public void testCaseTaxCreditWithEditedEvidenceReasonIsConfidentialityAppendix12DocHaveDocumentThenReject() {
+
+        callback.getCaseDetails().getCaseData().setDwpEditedEvidenceReason("childSupportConfidentiality");
+        callback.getCaseDetails().getCaseData().setAppendix12Doc(DwpResponseDocument.builder().documentLink(DocumentLink.builder().documentUrl("b.pdf").documentFilename("b.pdf").build()).build());
+        callback.getCaseDetails().getCaseData().getAppendix12Doc().setDocumentLink(DocumentLink.builder().documentUrl("b.pdf").documentFilename("b.pdf").build());
+
+        PreSubmitCallbackResponse<SscsCaseData> response = handler.handle(MID_EVENT, callback, USER_AUTHORISATION);
+
+
+        assertNotNull(response.getErrors());
```

Comment:
This is not really that useful - can you specify what errors you are expecting here otherwise could lead to issues being missed.